### PR TITLE
Update env use in gmsh package

### PR DESCRIPTION
**Pull request by cattabiani**

### CODE REVIEW

File: var/spack/repos/builtin/packages/gmsh/package.py [hunk 1]
```diff
@@ -144,3 +139,27 @@ def cmake_args(self):
...
+        if "+petsc" in self.spec:
+            env.prepend_path('PETSC_DIR', self.spec['petsc'].prefix)
+        if "+tetgen" in self.spec:
+            env.prepend_path('TETGEN_DIR', self.spec['tetgen'].prefix)
+        if "+netgen" in self.spec:
+            env.prepend_path('NETGEN_DIR', self.spec['netgen'].prefix)
+        if "+slepc" in self.spec:
+            env.prepend_path('SLEPC_DIR', self.spec['slepc'].prefix)
+
+    def setup_run_environment(self, env):
```
Comment (matz-e) [inline var/spack/repos/builtin/packages/gmsh/package.py]: Seems like this is only needed for CMake, so we would actually not need the variables in the runtime environment?

**Review by matz-e — COMMENTED** [review]
Thanks for taking care of this!

File: var/spack/repos/builtin/packages/gmsh/package.py (hunk 1 above)
Comment (cattabiani) [reply var/spack/repos/builtin/packages/gmsh/package.py]: so setup_run_environment should be removed? I am not sure I have the full picture here

**Comment by cattabiani** [comment]
please retest

File: var/spack/repos/builtin/packages/gmsh/package.py (hunk 1 above)
Comment (matz-e) [reply var/spack/repos/builtin/packages/gmsh/package.py]: I'd think so. Most of this stuff looks like it is for CMake, not actually needed when executing anything built with GMSH

File: var/spack/repos/builtin/packages/gmsh/package.py [hunk 2]
```diff
@@ -144,3 +139,27 @@ def cmake_args(self):
             options.append('-DENABLE_COMPRESSED_IO:BOOL=ON')
 
         return options
+
```
Comment (matz-e) [inline var/spack/repos/builtin/packages/gmsh/package.py]:
```suggestion

```

**Review by matz-e — DISMISSED** [review]

**Review by iomaganaris — APPROVED** [review]
LGTM

**Pull request merged**